File: src/easy/reverse_string.rs
pub fn reverse_string(s: &mut [char]) {
	if s.is_empty() {
		return;
	}
	let mut left = 0;
	let mut right = s.len().saturating_sub(1);
	while left < right {
		s.swap(left, right);
		left += 1;
		right -= 1;
	}

}

File: easy/reverse_string.go
package easy

func ReverseString(s []rune) {
	if len(s) == 0 {
		return
	}
	left, right := 0, len(s)-1
	var char rune
	for left < right {
		char = s[left]
		s[left] = s[right]
		s[right] = char
		left++
		right--
	}

}
